x-pack/plugins/apm/public/components/app/service_overview/service_overview_transactions_table/index.tsx

```diff
@@ -0,0 +1,329 @@
+        {
+          defaultMessage: 'Error rate',
+        }
+      ),
+      width: px(unit * 8),
+      render: (_, { error_rate: errorRate }) => {
+        return (
+          <SparkPlotWithValueLabel
+            color="euiColorVis7"
+            compact
+            series={errorRate.timeseries ?? undefined}
+            valueLabel={
+              errorRate.value !== null
+                ? asPercent(errorRate.value, 1)
+                : NOT_AVAILABLE_LABEL
+            }
+            start={parseFloat(start!)}
+            end={parseFloat(end!)}
+          />
+        );
+      },
+    },
+    {
+      field: 'impact',
+      name: i18n.translate(
+        'xpack.apm.serviceOverview.transactionsTableColumnImpact',
+        {
+          defaultMessage: 'Impact',
+        }
+      ),
+      width: px(unit * 4),
+      render: (_, { impact }) => {
+        return <ImpactBar value={impact ?? 0} size="m" />;
+      },
+    },
+  ];
+
+  return (
+    <EuiFlexGroup direction="column">
+      <EuiFlexItem>
+        <EuiFlexGroup justifyContent="spaceBetween">
+          <EuiFlexItem>
+            <EuiTitle size="xs">
+              <h2>
+                {i18n.translate(
+                  'xpack.apm.serviceOverview.transactionsTableTitle',
+                  {
+                    defaultMessage: 'Transactions',
+                  }
+                )}
+              </h2>
+            </EuiTitle>
+          </EuiFlexItem>
+          <TableLinkFlexItem>
+            <TransactionOverviewLink serviceName={serviceName}>
+              {i18n.translate(
+                'xpack.apm.serviceOverview.transactionsTableLinkText',
+                {
+                  defaultMessage: 'View transactions',
+                }
+              )}
+            </TransactionOverviewLink>
+          </TableLinkFlexItem>
+        </EuiFlexGroup>
+      </EuiFlexItem>
+      <EuiFlexItem>
+        <EuiFlexItem>
+          <TableFetchWrapper hasData={!!items.length} status={status}>
+            <EuiBasicTable
+              columns={columns}
+              items={items}
+              pagination={{
+                pageIndex,
+                pageSize: PAGE_SIZE,
+                totalItemCount,
+                pageSizeOptions: [PAGE_SIZE],
+                hidePerPageOptions: true,
+              }}
+              loading={status === FETCH_STATUS.LOADING}
+              onChange={(newTableOptions: {
+                page?: {
+                  index: number;
+                };
+                sort?: { field: string; direction: SortDirection };
+              }) => {
+                setTableOptions({
+                  pageIndex: newTableOptions.page?.index ?? 0,
+                  sort: newTableOptions.sort
+                    ? {
+                        field: newTableOptions.sort.field as SortField,
+                        direction: newTableOptions.sort.direction,
+                      }
+                    : DEFAULT_SORT,
```

Comment:
I suppose this is not possible?

```suggestion
                  sort: newTableOptions.sort ?? DEFAULT_SORT
```

Reply:
no, it will complain that `sortField` is a string (which is how we get it from `onChange`), instead of `SortField`, so we have to cast it as such. We could use the following, but not sure if that's more readable:

```ts
(newTableOptions.sort as
  | { field: SortField; direction: SortDirection }
  | undefined) || DEFAULT_SORT,
```

Reply:
I thought it was just a matter of changing lin 304:

```diff
-                sort?: { field: string; direction: SortDirection };
+                sort?: { SortField: string; direction: SortDirection };
```

But if that doesn't fix it I'm good with leaving it